src/main/java/training/busboard/UserInterface.java

```diff
@@ -0,0 +1,88 @@
+package training.busboard;
+
+import training.busboard.models.Bus;
+
+import java.util.InputMismatchException;
+import java.util.List;
+import java.util.Scanner;
+
+public class UserInterface {
+    private Scanner scanner = new Scanner(System.in);
+
+    public void welcome() {
+        System.out.println("Welcome to BusBoard!!");
+        System.out.println("=====================");
+    }
+
+    public String getServiceToRun() {
+        System.out.println("Would you like to search for buses by postcode (enter p) or by stopcode (enter s)?");
+        System.out.print("Enter /q/ to quit. > ");
+        String input = scanner.nextLine();
+        try {
+            return this.processInput(input, "[sp]");
+        } catch (InputMismatchException e) {
+            System.out.println("You have entered an invalid postcode. Please try again.");
+            return this.getPostcode();
+        }
+    }
+
+    public String getStopcode() {
+        System.out.print("Please enter a stop code or enter /q/ to quit. > ");
+        String input = scanner.nextLine().trim().toLowerCase();
+        try {
+            return this.processInput(input, "\\d+\\w*");
+        } catch (InputMismatchException e) {
+            System.out.println("You have entered an invalid stopcode. Please try again.");
```

Comment:
This error message needs updating :)

Reply:
How so?

Reply:
Whoops - clicked on the wrong line of code. New comment appearing in a few seconds...